internal/setup/setup.go

```diff
@@ -64,6 +64,18 @@ const (
 	GlobPath    PathKind = "glob"
 	TextPath    PathKind = "text"
 	SymlinkPath PathKind = "symlink"
+	// GeneratePath is a special kind of globs with the following format:
+	//   /absolute/explicit/path/to/dir/**
+	// Thus, the path /foo/bar/** is valid while the folllowing are not:
+	//   ../foo/bar/**
+	//   /fo*/bar/**
+	//   /foo/bar/
+	//   /foo/bar
+	// There are only two possible values of "generate" as of now:
+	//   "" (empty)
+	//   "chisel-state"
+	// Empty value signifies that the path is not a GeneratePath.
```

Comment:
I hate to discourage documentation but I am not sure this is the best place because the other attributes are not documented here. Let's think about it.

Reply:
I would still want to keep it regardless if the others are documented or not. To be honest, I think the others should be documented too. Let me know if you think there are any better places for the new comment however.